File: Task/royalscouts_backend/src/main/java/com/project/royalscouts/models/Review.java
package com.project.royalscouts.models;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Table(name = "reviews")
public class Review  extends AuditModel {

  @Column(name = "reviewed_by")
  private String reviewedBy;

  @Column(name = "comments")
  private String comments;

  @Column(name = "permission_action")
  private String permissionAction;

  @Column(name = "frequency")
  private String frequency;

  @ManyToOne(fetch = FetchType.LAZY, optional = false)
  @JoinColumn(name = "task_id", nullable = false)
  @OnDelete(action = OnDeleteAction.CASCADE)
  @JsonIgnore
  private Task task;

  public Review() {
  }

  public String getReviewedBy() {
    return reviewedBy;
  }

  public void setReviewedBy(String reviewedBy) {
    this.reviewedBy = reviewedBy;
  }

  public String getComments() {
    return comments;
  }

  public void setComments(String comments) {
    this.comments = comments;
  }

  public String getPermissionAction() {
    return permissionAction;
  }

  public void setPermissionAction(String permissionAction) {
    this.permissionAction = permissionAction;
  }

  public String getFrequency() {
    return frequency;
  }

  public void setFrequency(String frequency) {
    this.frequency = frequency;
  }

  public Task getTask() {
    return task;
  }

  public void setTask(Task task) {
    this.task = task;
  }

  

}

File: Task/royalscouts_frontend/lib/app/core/models/review.dart
class Review {
  final num id;
  final String comments;
  final String reviewedBy;
  final String permissionAction;
  final String frequency;
  final String createdAt;

  Review({
    required this.id,
    required this.comments,
    required this.reviewedBy,
    required this.permissionAction,
    required this.frequency,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      comments: json['comments'],
      reviewedBy: json['reviewedBy'],
      permissionAction: json['permissionAction'],
      frequency: json['frequency'],
      createdAt: json['createdAt']
    );
  }
}
